src/DDM.jl

```diff
@@ -381,67 +383,325 @@ sampler(rng::AbstractRNG, d::DDM) = rand(rng::AbstractRNG, d::DDM)
+#         end
+#     else #sz=$
+#         if st==0 #sv=0,sz=$,st=0
+#             return _simpson_1D(rt, ν, η, α, z, τ, ϵ, z-sz/2., z+sz/2., n_sz, τ, τ , 0)
+#         else     #sv=0,sz=$,st=$
+#             return _simpson_2D(rt, ν, η, α, z, τ, ϵ, z-sz/2., z+sz/2., n_sz, τ-st/2., τ+st/2., n_st)
+#     end
+# end
+
+# """
+
+# _simpson_1D(x::Real, ν::Real, η::Real, α::Real, z::Real, τ::Real, ϵ::::Real, lb_z::Real, ub_z::Real, n_sz::Int, lb_t::Real, ub_t::Real, n_st::Int)
+
+#     Numerical Integration with Simpson's Method
+#     see: https://en.wikipedia.org/wiki/Simpson%27s_rule
+
+# # Arguments
+# - `x`::Real: response time
+# - `ν`::Real: response time
+# - `η`::Real: response time
+# - `α`::Real: response time
+# - `z`::Real: response time
+# - `τ`::Real: response time
+# - `ϵ`::Real: response time
+# - `lb_z`::Real: response time
+# - `ub_z`::Real: response time
+# - `n_sz`::Int: response time
+# - `lb_t`::Real: response time
+# - `ub_t`::Real: response time
+# - `n_st`::Int: response time
+
+# """
+# function _simpson_1D(x::Real, ν::Real, η::Real, α::Real, z::Real, τ::Real, ϵ::::Real, lb_z::Real, ub_z::Real, n_sz::Int, lb_t::Real, ub_t::Real, n_st::Int)
+#         #assert ((n_sz&1)==0 and (n_st&1)==0), "n_st and n_sz have to be even"
+
+# end
+# """
+#     RatcliffDDM 
+
+#     Model object for the Ratcliff Diffusion Model.
 
-# Constructor for the Ratcliff Diffusion Decision Model. 
+# # Fields
+#     - `ν`: drift rate. Average slope of the information accumulation process. The drift gives information about the speed and direction of the accumulation of information. Typical range: -5 < ν < 5
+#     - `α`: boundary threshold separation. The amount of information that is considered for a decision. Typical range: 0.5 < α < 2
+#     - `τ`: non-decision time. The duration for a non-decisional processes (encoding and response execution). Typical range: 0.1 < τ < 0.5 
+#     - `z`: starting point. Indicator of an an initial bias towards a decision. The z parameter is relative to a (i.e. it ranges from 0 to 1).
+#     - `η`:  across-trial-variability of drift rate. Typical range: 0 < η < 2. Default is 0.
+#     - `sz`: across-trial-variability of starting point. Typical range: 0 < sz < 0.5. Default is 0.
+#     - `st`: across-trial-variability of non-decision time. Typical range: 0 < st < 0.2. Default is 0.
+#     - `σ`: diffusion noise constant. Default is 1.
+
+# # Example 
+
+# ````julia
+# using SequentialSamplingModels
+# dist = RatcliffDDM(ν = 0.50,α = 0.08,τ = 0.30,z = 0.04,η = 0.10,sz = 0.02,st = .02,σ = 0.10) 
+# choice,rt = rand(dist, 10)
+# like = pdf.(dist, choice, rt)
+# loglike = logpdf.(dist, choice, rt)
+# ````
+    
+# # References
+        
+# Ratcliff, R., & McKoon, G. (2008). The Diffusion Decision Model: Theory and Data for Two-Choice Decision Tasks. Neural Computation, 20(4), 873–922.
+
+# Ratcliff, R. (1978). A theory of memory retrieval. Psychological Review, 85, 59–108. https://doi.org/10.1037/0033-295X.85.2.59
+
+# """
```

Comment:
FYI, you can remove the @concrete since you are specifying parametric types.

Reply:
Thanks!